addon/validations/factory.js

```diff
@@ -454,6 +462,33 @@ function getCPDependentKeysFor(attribute, validations, owner) {
 }
 
 /**
+ * Extract all dependentKeys from any property that is a CP
+ *
+ * @method extractOptionsDependentKeys
+ * @param  {Object} options
+ * @return {Array}  dependentKeys
+ */
+function extractOptionsDependentKeys(options) {
+  if(options && typeof options === 'object') {
+    return Object.keys(options).reduce((arr, key) => {
+      let option = options[key];
+
+      if(typeof option === 'object' && option.isDescriptor) {
+        let dependentKeys = option._dependentKeys || [];
```

Comment:
I'm pretty sure the `dependentKeys` option (and henceforth, extraction) is a thing from the past with this PR. I need a bit more explanation on why it's still required. (I see where it's used, I think we can get rid of it)